Update documentations of Configuration default environments (#3321)
kedro-org/kedro: docs/source/configuration/advanced_configuration.md

By default, Kedro is set up to use the [OmegaConfigLoader](/kedro.config.OmegaConfigLoader) class.

> ## Advanced Configuration for Kedro Project

Reviewer: ```suggestion
## Advanced configuration for Kedro projects
```

### How to bypass the configuration loading rules
You can bypass the configuration patterns and set configuration directly on the instance of a config loader class. You can bypass the default configuration (catalog, parameters and credentials) as well as additional configuration.

> For example, you can [use hooks to load external credentials](../hooks/common_use_cases.md##use-hooks-to-load-external-credentials).

Reviewer: ```suggestion
For example, you can [use hooks to load external credentials](../hooks/common_use_cases.md#use-hooks-to-load-external-credentials).
```

> Alternatively, if you are using config loader as a standalone component, you can override configuration as follow:

Reviewer 2: ```suggestion
Alternatively, if you are using config loader as a standalone component, you can override configuration as follows:
```

If no merge strategy is defined, the default destructive strategy will be applied. Note that this merge strategy setting only applies to configuration files in **different** environments.
When files are part of the same environment, they are always merged in a soft way. An error is thrown when files in the same environment contain the same top-level keys.

> ## Advance Configurations without Kedro Project

Reviewer: ```suggestion
## Advanced configuration without a full Kedro project
```

> By defaults, Kedro Project has a `base` and `local` environments. In some cases, you may only want to use the `OmegaConfigLoader` without a Kedro project. This is possible because Kedro embrace [modularity at the core](https://github.com/kedro-org/kedro/wiki/Kedro-Principles#1-modularity-at-the-core-%EF%B8%8F).

Reviewer: ```suggestion
By default, a Kedro project has a `base` and `local` environment. In some cases, you may only want to use the `OmegaConfigLoader` without a Kedro project. This is possible because Kedro embraces [modularity at the core](https://github.com/kedro-org/kedro/wiki/Kedro-Principles#1-modularity-at-the-core-%EF%B8%8F).
```

Reviewer: I think this probably needs a bit more attention. First, I'd suggest we reorder the argument so it's presented as:

* In some cases you may want to use the config loader without a full Kedro project (with a link to docs about this or further explanation of the use cases that mean you'd want to do it and where/how to do it)
* But by default there's an expectation of `base` and `local`

I'd remove the sentence about modularity.

However, when you use the `OmegaConfigLoader` directly, it assumes *no* environment. You may find it useful to [add Kedro to your existing notebooks](../notebooks_and_ipython/notebook-example/add_kedro_to_a_notebook.md).

### Read configuration
The config loader can work without a Kedro project structure.
```bash
tree .
.
└── parameters.yml
```

```yaml
# parameters.yml
learning_rate: 0.01
train_test_ratio: 0.7
```

```python
from kedro.config import OmegaConfigLoader
config_loader = OmegaConfigLoader(conf_source=".")

# Optionally, you can also use environments
# config_loader = OmegaConfigLoader(conf_source=".", base_env="base", default_run_env="local")

>>> config_loader["parameters"]
{'learning_rate': 0.01, 'train_test_ratio': 0.7}
```

For the full list of features, please refer to [configuration_basics](./configuration_basics.md) and [advanced_configuration](./advanced_configuration.md)

### How to use Custom Resolvers with `OmegaConfigLoader`
You can register custom resolvers to use non-primitive types for parmaeters.

```yaml
# parameters.yml

> polar_float64: "${polars: Float64}"

Reviewer 3: ```suggestion
polars_float64: "${polars:Float64}"
```

today: "${today:}"
```

```python
import polars as pl
from datetime import date

from kedro.config import OmegaConfigLoader

custom_resolvers = {"polars": lambda x: getattr(pl, x),
                    "today": lambda: date.today()}

# Register custom resolvers
config_loader = OmegaConfigLoader(conf_source=".", custom_resolvers=custom_resolvers)
>>> print(config_loader["parameters"])

> {'polar_float64': Float64, 'today': datetime.date(2023, 11, 23)}

Reviewer 3: ```suggestion
{'polars_float64': Float64, 'today': datetime.date(2023, 11, 23)}
```

If no merge strategy is defined, the default destructive strategy will be applied. Note that this merge strategy setting only applies to configuration files in **different** environments.
When files are part of the same environment, they are always merged in a soft way. An error is thrown when files in the same environment contain the same top-level keys.

## Advanced configuration without a full Kedro project

>  In some cases, you may only want to use the `OmegaConfigLoader` without a Kedro project. By default, a Kedro project have a `base` and `local` environment.

Reviewer 2: Is it without a Kedro project or without the full Kedro framework? 

```suggestion
 In some cases, you may only want to use the `OmegaConfigLoader` without a Kedro project. By default, a Kedro project has a `base` and `local` environment.
```

> 

Reviewer 2: ```suggestion
```

> Unlike a Kedro Project, when you use the `OmegaConfigLoader` directly, it assumes *no* environment. You may find it useful to [add Kedro to your existing notebooks](../notebooks_and_ipython/notebook-example/add_kedro_to_a_notebook.md).

Reviewer 2: ```suggestion
However, when you use the `OmegaConfigLoader` directly, it assumes *no* environment. You may find it useful to [add Kedro to your existing notebooks](../notebooks_and_ipython/notebook-example/add_kedro_to_a_notebook.md).
```

### Read configuration
The config loader can work without a Kedro project structure.
```bash
tree .
.
└── parameters.yml
```

```yaml
# parameters.yml
learning_rate: 0.01
train_test_ratio: 0.7
```

```python
from kedro.config import OmegaConfigLoader
config_loader = OmegaConfigLoader(conf_source=".")

# Optionally, you can also use environments
# config_loader = OmegaConfigLoader(conf_source=".", base_env="base", default_run_env="local")

>>> config_loader["parameters"]
{'learning_rate': 0.01, 'train_test_ratio': 0.7}
```

For the full list of features, please refer to [configuration_basics](./configuration_basics.md) and [advanced_configuration](./advanced_configuration.md)

### How to use Custom Resolvers with `OmegaConfigLoader`

> You can register custom resolvers to use non-primitive type for parmaeters.

Reviewer 2: ```suggestion
You can register custom resolvers to use non-primitive types for parameters.
```
